autoPyTorch/api/base_task.py

```diff
@@ -1237,24 +1384,119 @@ def fit(self,
+                provided, uses the default in the pipeline config
+            pipeline_options (Optional[Dict]):
+                Valid config options include "device",
+                "torch_num_threads", "early_stopping", "use_tensorboard_logger",
+                "metrics_during_training"
+            disable_file_output (Optional[List[Union[str, DisableFileOutputParameters]]]):
+                Used as a list to pass more fine-grained
+                information on what to save. Must be a member of `DisableFileOutputParameters`.
+                Allowed elements in the list are:
+
+                + `y_optimization`:
+                    do not save the predictions for the optimization set,
+                    which would later on be used to build an ensemble. Note that SMAC
+                    optimizes a metric evaluated on the optimization set.
+                + `pipeline`:
+                    do not save any individual pipeline files
+                + `pipelines`:
+                    In case of cross validation, disables saving the joint model of the
+                    pipelines fit on each fold.
+                + `y_test`:
+                    do not save the predictions for the test set.
+                + `all`:
+                    do not save any of the above.
+                For more information check `autoPyTorch.evaluation.utils.DisableFileOutputParameters`.
+            configuration: (Configuration)
+                configuration to fit the pipeline with.
 
         Returns:
-            BasePipeline:
+            (BasePipeline):
                 fitted pipeline
+            (RunInfo):
+                Run information
+            (RunValue):
+                Result of fitting the pipeline
+            (BaseDataset):
+                Dataset created from the given tensors
         """
-        self.dataset_name = dataset.dataset_name
 
-        if self._logger is None:
-            self._logger = self._get_logger(str(self.dataset_name))
+        if dataset is None:
+            assert X_train is not None and \
+                   y_train is not None, "No dataset provided, must provide X_train, y_train tensors"
+            dataset = self.get_dataset(X_train=X_train,
+                                       y_train=y_train,
+                                       X_test=X_test,
+                                       y_test=y_test,
+                                       resampling_strategy=resampling_strategy,
+                                       resampling_strategy_args=resampling_strategy_args,
+                                       dataset_name=dataset_name
+                                       )
+
+        # TAE expects each configuration to have a config_id.
+        # For fitting a pipeline as it is not part of the
+        # search process, it makes sense to set it to 0
+        if hasattr(configuration, 'config_id') or configuration.config_id is None:
```

Comment:
or, but not and?
I do not get why it makes sense to overwrite `config_id`.

Reply:
`config_id` is updated inside the `ConfigSpace.Configuration` class when `smac` queries a new config. However, in this case the configuration is coming from the user and it does not have a config id so we need to set it. In the case of fitting a pipeline we really dont care what the config id is, so setting it `0` seems fine to me.

Reply:
I got what you meant in the comment out now 
Then we don't need the `if` itself?
Or does that raise any error?

Reply:
Actually, I dont think we need the `if` statement. I'll try the tests without the `if` statement and if its not necessary I'll remove it.

Reply:
Thanks, it will look more clear to me